clang/include/clang/CIR/Dialect/IR/CIRTypes.td

```diff
@@ -266,6 +266,22 @@ def CIR_PointerType : CIR_Type<"Pointer", "ptr",
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// BoolType
+//===----------------------------------------------------------------------===//
+
+def CIR_BoolType :
+    CIR_Type<"Bool", "bool",
+             [DeclareTypeInterfaceMethods<DataLayoutTypeInterface>]> {
+
+  let summary = "CIR bool type";
+  let description = [{
+    `cir.bool` represents C++ bool type.
+  }];
+
+  let hasCustomAssemblyFormat = 1;
```

Comment:
Custom assembly format and related `BoolType::print`, `BoolType::parse` are unnecessary here as default printer/parser generates `!cir.bool` anyway.

I cleaned this in incubator in https://github.com/llvm/clangir/pull/1407